opentreemap/works_management/models.py

```diff
@@ -21,10 +22,28 @@ class Team(models.Model):
 class WorkOrder(Auditable, models.Model):
     instance = models.ForeignKey(Instance)
     name = models.CharField(max_length=255, null=False, blank=False)
+    reference_num = models.IntegerField(default=0)
     created_at = models.DateTimeField(auto_now_add=True)
     created_by = models.ForeignKey(User)
     updated_at = models.DateTimeField(auto_now=True)
 
+    class Meta:
+        unique_together = ('instance', 'reference_num')
+
+    def get_next_reference_num(self):
+        """
+        Return next sequential reference number for instance.
+        """
+        agg_result = Instance.objects.filter(id=self.instance.id) \
+            .annotate(max_value=Max('workorder__reference_num'))[0]
+        max_value = agg_result.max_value or 0
+        return max_value + 1
```

Comment:
We could avoid using an aggregate query by adding `workorder_sequence` and `task_sequence` columns to `treemap_instance` table. Atomically incrementing these would behave in a similar way to using lower level Postgres sequences.